app/queries/facility_analytics_query.rb

```diff
@@ -54,6 +55,46 @@ def registered_patients_by_period
     group_by_user_and_date(@registered_patients_by_period, :registered_patients_by_period)
   end
 
+  def follow_up_patients_by_period
+    @follow_up_patients_by_period ||=
+      #
+      # The DISTINCT ON is for when a patient visits a facility twice in a period and is seen by different users.
+      # Without it, the visit is counted twice and the total increases in comparison to facility follow ups.
+      #
+      # The salient clause: we're DISTINCTing ON the BloodPressure.date_to_period_sql and the patient_id.
+      #
+      # However, we've turned this into a nested subquery using .from as a workaround for ActiveRecord's inability to
+      # compose a COUNT with a DISTINCT ON – it ends up with garble like COUNT DISTINCT DISTINCT ON which is a
+      # completely invalid query.
+      #
+      Patient
+        .from(Patient
+                .joins(:blood_pressures)
+                .hypertension_follow_ups_by_period(@period, last: @prev_periods)
+                .distinct(false) # this removes the distinct from hypertension_follow_ups so we can apply DISTINCT ON
+                .group("bp_user_id",
+                  "blood_pressures.patient_id",
+                  BloodPressure.date_to_period_sql("blood_pressures.recorded_at", @period),
+                  "blood_pressures.recorded_at",
+                  "patients.deleted_at")
+                .where(blood_pressures: {facility: @facility})
+                .select(%(
+                    DISTINCT ON (blood_pressures.patient_id,
+                    #{BloodPressure.date_to_period_sql("blood_pressures.recorded_at", @period)})
+                    blood_pressures.user_id AS bp_user_id,
+                    patients.deleted_at))
+                .select("blood_pressures.recorded_at AS bp_recorded_at")
+                .order("blood_pressures.patient_id",
+                  Arel.sql(BloodPressure.date_to_period_sql("blood_pressures.recorded_at", @period)),
+                  "blood_pressures.recorded_at"),
+          "patients")
+        .group("bp_user_id")
+        .group_by_period(:month, "bp_recorded_at")
+        .count
+
+    group_by_user_and_date(@follow_up_patients_by_period, :follow_up_patients_by_period)
```

Comment:
Is this any different from the query in the districts query object? At first glance they appear the same, which makes me think it should be its own object that can just work with a Region.

Reply:
The facility query does a lot of grouping by _user_, whereas the district query groups by _facility_. It's a bit awkward to detangle them cleanly :/

Reply:
Hmm, perhaps we _could_ pull the base query into a separate object and leave the grouping and data structure massaging in here. I'll take a stab at it and see if it turns out ok.

Reply:
@rsanheim the queries are practically the same. The difference is that the districts query groups results by facility and the facility query groups results by user. If you still think it should work with a Region, I'd appreciate your guidance on how to make that change